python/triton/compiler/code_generator.py

```diff
@@ -506,6 +520,15 @@ def visit_List(self, node):
         elts = language.tuple([self.visit(elt) for elt in node.elts])
         return elts
 
+    def emit_return(self, handles):
+        for i, (name, ty) in enumerate(self.prototype.args_mut):
+            value = self.dereference_name(name)
```

Comment:
Seems wrong. We should track the python object, not the value the name is currently bound to. e.g.
```python
def f(a):
    a = NewAggregate(...)
```

should not mutate the caller's `a` value.